extensions/git/src/repository.ts

```diff
@@ -857,12 +857,12 @@ export class Repository implements Disposable {
 		outputChannel: OutputChannel,
 		private telemetryReporter: TelemetryReporter
 	) {
-		const workspaceWatcher = workspace.createFileSystemWatcher('**');
+		const rootUri = Uri.file(repository.root);
+		const workspaceWatcher = workspace.createFileSystemWatcher(new RelativePattern(rootUri, '**'));
 		this.disposables.push(workspaceWatcher);
 
-		const onWorkspaceFileChange = anyEvent(workspaceWatcher.onDidChange, workspaceWatcher.onDidCreate, workspaceWatcher.onDidDelete);
-		const onWorkspaceRepositoryFileChange = filterEvent(onWorkspaceFileChange, uri => isDescendant(repository.root, uri.fsPath));
-		const onWorkspaceWorkingTreeFileChange = filterEvent(onWorkspaceRepositoryFileChange, uri => !/\/\.git($|\/)/.test(uri.path));
+		const onWorkspaceRepositoryFileChange = anyEvent(workspaceWatcher.onDidChange, workspaceWatcher.onDidCreate, workspaceWatcher.onDidDelete);
+		const onWorkspaceWorkingTreeFileChange = filterEvent(onWorkspaceRepositoryFileChange, uri => !/\/\.git($|\/)/.test(uri.path.substr(rootUri.path.length + 1)));
```

Comment:
@eamodio, sorry for not getting to this sooner. Is there an "off by one bug" here? This regular expression matches whether the string ends with `/.git` or it contains `/.git/`. Using `rootUri.path.length + 1` strips off the starting `/`. Am I missing something here?

Reply:
Yeah we probably need a better way to get the _relative path to the root URI_ than substr magic.

Reply:
There is a helper function in the git extension but at the end of the day that is also using `substring`.
https://github.com/microsoft/vscode/blob/7d0277ac210b4091da6c85d97ff41409e117eac6/extensions/git/src/util.ts#L322